### PR TITLE
Task_1_1_1

### PR DESCRIPTION
**Pull request by TheAndrewLa**
First version of 'heapsort' solution

### CODE REVIEW

File: Task_1_1_1/src/test/java/SortingTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        return true;
+    }
+
+    // General test (random array)
+
+    @Test
+    void checkRandomArray() {
+        Random rand = new Random();
+
+        for (int length = 1; length < 100; length += 10) {
```
Comment (Sharapov-Maksim) [inline Task_1_1_1/src/test/java/SortingTest.java]: Тестировать сортировку на массивах разных размеров это правильно. Однако такие циклы с итеративным изменением размера по моему избыточны. Например, запуски с `length=61` и `length=71` вряд ли приведут к разным результатам. А вот запуски с `length=` `0, 1, 2, 10, 10000` уже кажутся более разнообразными. Но лучше всё это разбивать на разные тесты, т.к. при падении маленького теста его проще дебажить

File: Task_1_1_1/src/test/java/SortingTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    // Some corner-cases
+    // (sorted, reverse-sorted and array with only one number (e.g. {1, 1, 1...}) )
+
+    @Test
+    void checkSortedArray() {
+        Random rand = new Random();
+
+        for (int length = 1; length < 100; length += 10) {
```
Comment (Sharapov-Maksim) [inline Task_1_1_1/src/test/java/SortingTest.java]: `100` и `10` - неименованные константы, такие стоит выделять в `static final` поля класса

File: Task_1_1_1/src/test/java/SortingTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            int[] array = new int[length];
+            int base = rand.nextInt(100);
```
Comment (Sharapov-Maksim) [inline Task_1_1_1/src/test/java/SortingTest.java]: А эта `100` уже по смыслу другая, поэтому и ей нужно дать своё имя

**Review by Sharapov-Maksim — CHANGES_REQUESTED** [review]
Добавил замечания, комментируйте, исправляйте, запрашивайте следущую итерацию ревью

**Review by Sharapov-Maksim — APPROVED** [review]
